src/coins.h

```diff
@@ -297,12 +297,22 @@ class CCoinsViewCache : public CCoinsViewBacked
     bool SpendCoin(const COutPoint &outpoint, Coin* moveto = nullptr);
 
     /**
-     * Push the modifications applied to this cache to its base.
-     * Failure to call this method before destruction will cause the changes to be forgotten.
+     * Push the modifications applied to this cache to its base and wipe local state.
+     * Failure to call this method or Sync() before destruction will cause the changes
+     * to be forgotten.
      * If false is returned, the state of this cache (and its backing view) will be undefined.
      */
     bool Flush();
 
+    /**
+     * Push the modifications applied to this cache to its base while retaining
+     * the contents of this cache (except for FRESH spent coins, which we erase).
```

Comment:
I don't think `except for FRESH spent coins, which we erase` is true?

Reply:
Yep, [you're right](https://github.com/jamesob/bitcoin/blob/0292b01cc4be18e5eb37563064cced7d73cd7025/src/coins.cpp#L120-L121), not sure what I was thinking there.